### ISSUE TITLE
Falsy values cause validation exception

### ISSUE THREAD

**Issue by garth** [prisma/prisma#14153]
### Bug description

When the where clause contains falsy data the prisma validation blocks the query from running even when the query/data is valid.

```
Error: Cannot create Prisma Client where unique input because the source data ({ meeti ngId: 1234, userId: 0 }) is missing the following unique identifier fields: userId
```

### How to reproduce

```ts
      prisma.meetingStaff.findUnique({
        where: {
          meetingId_userId: {
            meetingId: 1234,
            userId: 0,
          }
        }
      })
```

### Expected behavior

0 is a valid value, query should be executed

### Prisma information

```
model Meeting {
  id                         Int                         @id @default(autoincrement())
  title                      String?
  meetingStaff               MeetingStaff[]

  @@map("meeting")
}

model MeetingStaff {
  meetingId Int     @map("meeting")
  userId    Int     @map("usernr")
  meeting   Meeting @relation(fields: [meetingId], references: [id], onDelete: Cascade)
  user      User    @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@id([meetingId, userId])
  @@map("meeting_staff")
}

model User {
  id                          Int                          @id @default(autoincrement())
  firstName                   String?                      @map("firstname")
  lastName                    String?                      @map("surname")
  meetingStaff                MeetingStaff[]

  @@map("users")
}
```

### Environment & setup

- OS: Mac OS
- Database: PostgreSQL
- Node.js version: v16.15.1

### Prisma Version

```
prisma                  : 4.0.0
@prisma/client          : 4.0.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine da41d2bb3406da22087b849f0e911199ba4fbf11 (at node_modules/.pnpm/@prisma+engines@3.16.0-49.da41d2bb3406da22087b849f0e911199ba4fbf11/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli da41d2bb3406da22087b849f0e911199ba4fbf11 (at node_modules/.pnpm/@prisma+engines@3.16.0-49.da41d2bb3406da22087b849f0e911199ba4fbf11/node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core da41d2bb3406da22087b849f0e911199ba4fbf11 (at node_modules/.pnpm/@prisma+engines@3.16.0-49.da41d2bb3406da22087b849f0e911199ba4fbf11/node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt da41d2bb3406da22087b849f0e911199ba4fbf11 (at node_modules/.pnpm/@prisma+engines@3.16.0-49.da41d2bb3406da22087b849f0e911199ba4fbf11/node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Default Engines Hash    : da41d2bb3406da22087b849f0e911199ba4fbf11
Studio                  : 0.465.0
```

**Comment by aqrln** [comment]
I couldn't reproduce it with 4.0.0 too. Any chance you might provide a clonable repo with a reproduction? This looks like a bad bug, but it looks like there's more to it than just the value being falsy, although I copied your schema and the query verbatim, so I'm not quite sure what might differ.

**Comment by garrensmith** [comment]
@garth I'm going to close this ticket since we cannot reproduce it. If this is still an issue please open it again.

**Comment by janpio** [comment]
@garth This is how my GitHub is displaying your error message:
<img width="317" alt="image" src="https://user-images.githubusercontent.com/183673/183592831-9acb8b05-729d-4767-a0e8-cfb0f2cb7435.png">
Can you maybe retype `meetingId` in your Prisma Client Query to make sure there are not invisible characters in between that could problems? Then everything should just work.

**Comment by garth** [comment]
Thanks everyone for looking into this. We are working with 20+ year old data, so we sometimes find some odd behaviours, particularly when working with ts/js.  We also had problems with other 3rd party libs that use `if (row.id) {}` or similar logic and submitted a few PRs here and there. But in the end we decided the best option would be to "fix" the data and we added migrations to update all primary keys that are not truthy values.
